src/test/java/emissary/util/io/LoggingPrintStreamTest.java

```diff
@@ -0,0 +1,305 @@
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static emissary.util.io.LoggingPrintStream.NORMAL_SEPARATOR;
+import static emissary.util.io.LoggingPrintStream.THROWABLE_PREFIX;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class LoggingPrintStreamTest {
+    private static final String LOG_MSG_1 = "Log message NUMBER 1";
+    private static final String LOG_MSG_2 = "Log message NUMBER 2";
+    private static final Object LOG_OBJ_1 = Integer.valueOf(1234567890);
+    private static final Exception EXCEPTION_CAUSE = new Exception("Test Exception 'cause'");
+    private static final Exception EXCEPTION_ONE = new Exception("Test Exception NUMBER 1", EXCEPTION_CAUSE);
+    private static final Exception EXCEPTION_TWO = new Exception("Test Exception NUMBER 2", EXCEPTION_CAUSE);
+
+    @Test
+    void testArguments() {
+        final Logger logger = (Logger) LoggerFactory.getLogger(LoggingPrintStream.class);
+        final org.slf4j.event.Level slf4jLevel = org.slf4j.event.Level.TRACE;
+
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(null, "TEST", logger, slf4jLevel, 1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, null, logger, slf4jLevel, 1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, "TEST", null, slf4jLevel, 1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, "TEST", logger, null, 1, TimeUnit.SECONDS));
+        assertThrows(IllegalArgumentException.class, () -> new LoggingPrintStream(System.out, "TEST", logger, slf4jLevel, -1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, "TEST", logger, slf4jLevel, 1, null));
+    }
+
+    @Test
+    void testClose() throws Exception {
+        final Logger logger = (Logger) LoggerFactory.getLogger(LoggingPrintStream.class);
+        final org.slf4j.event.Level slf4jLevel = org.slf4j.event.Level.TRACE;
+
+        try (LoggingPrintStream loggingPrintStream = new LoggingPrintStream(System.out, "TEST", logger, slf4jLevel, 30, TimeUnit.SECONDS)) {
+            loggingPrintStream.close();
+
+            assertThrows(RejectedExecutionException.class, () -> loggingPrintStream.println(LOG_MSG_1));
+            assertThrows(RejectedExecutionException.class, () -> EXCEPTION_ONE.printStackTrace(loggingPrintStream));
+        }
+    }
+
+    @Test
+    void testLoggingLevel() throws Exception {
+        try (LogbackTester logbackTester = new LogbackTester(LoggingPrintStreamTest.class.getName());
+                LoggingPrintStream loggingPrintStreamDebug = new LoggingPrintStream(System.out, logbackTester.name + "_DEBUG", logbackTester.logger,
```

Comment:
Please wrap a different OutputStream so the tests don't abort in IntelliJ.  Can use `System.err`, a `ByteArrayOutputStream()`, `NullOutputStream.NULL_OUTPUT_STREAM`, whatever, but closing `System.out` causes problems.